hi-ml-azure/src/health_azure/utils.py

```diff
@@ -791,7 +892,30 @@ def merge_conda_files(files: List[Path], result_file: Path) -> None:
         unified_definition[CHANNELS] = channels
 
     try:
-        deps = conda_merge.merge_dependencies(env.get(DEPENDENCIES) for env in env_definitions)
+        deps_to_merge = [env.get(DEPENDENCIES) for env in env_definitions]
+        if len(extra_pip_deps) > 0:
+            deps_to_merge.extend([[{"pip": extra_pip_deps}]])
+        deps = conda_merge.merge_dependencies(deps_to_merge)
+
+        # Remove duplicated pip packages from merged dependencies sections. Note that for a package that is
+        # duplicated, the first value encountered will be retained.
+        pip_deps_entry = [d for d in deps if isinstance(d, dict) and "pip" in d][0]
```

Comment:
This silent restriction on "first value encountered" sounds dangerous. Can we print those out as logging.warning if the versions of duplicate packages do not match?

Reply:
Done. And I have added a parameter to override which one should be kept (at the moment just the first one it encounters, but we could extend this to keep the last, highest version etc)